dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -49,24 +60,115 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterF
 {
     auto lock = m_lock.lock_shared();
     RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
-    
+
+    // To be completed after Sharath's PR
+
     return E_NOTIMPL;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterForegroundActivator(_In_ IWpnForegroundSink* sink, _In_ PCWSTR processName)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterActivator(_In_ PCWSTR processName) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+    RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
+
+    std::wstring appId = GetAppIdentifier(processName);
+    m_notificationListenerManager.AddListener(appId, processName);
+
+    if (!m_notificationListenerManager.IsEmpty())
+    {
+        m_lifetimeManager.Cancel();
+    }
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::UnregisterActivator(_In_ PCWSTR processName) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+    RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
+
+    m_notificationListenerManager.RemoveListener(processName);
+    m_foregroundSinkManager->Remove(processName);
+
+    // TODO: Remove the app from storage
+
+    if (m_notificationListenerManager.IsEmpty())
+    {
+        m_lifetimeManager.Setup();
+    }
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterForegroundActivator(_In_ IWpnForegroundSink* sink, _In_ PCWSTR processName) noexcept try
 {
     auto lock = m_lock.lock_exclusive();
     RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
 
-    m_foregroundSinkManager.Add(processName, sink);
+    std::wstring appId = GetAppIdentifier(processName);
+
+    m_notificationListenerManager.AddListener(appId, processName);
+    m_foregroundSinkManager->Add(processName, sink);
```

Comment:
@pmpurifoy  Why do we use processname as the currency for ForegroundSinkManager instead of appId? Is there a reason for that? @danielayala94 Why have 2 different currencies?

Reply:
At the time the foreground sink was introduced, there wasn't any need of using AppId as the sink ID. Also, if from the SDK we get processName, we can directly use it to do any operation in the foreground sink manager.

But, in the big picture there is no consistency: the appId is the key (therefore, ID) in the storage and the listener ID too. I'm going to change the foreground sink manager ID to be appId.

Reply:
Sounds like a plan. We shouldn't use processName anywhere but at the entry points of LRP Platform. Always convert to appId and use that as the currency in internal components.